memcached.c

```diff
@@ -2961,6 +3008,24 @@ enum store_item_type do_store_item(item *it, int comm, conn *c, const uint32_t h
             c->thread->stats.slab_stats[ITEM_clsid(old_it)].cas_hits++;
             pthread_mutex_unlock(&c->thread->stats.mutex);
 
+            STORAGE_delete(c->thread->storage, old_it);
+            item_replace(old_it, it, hv);
+            stored = STORED;
+        } else if (c->set_stale && ITEM_get_cas(it) < ITEM_get_cas(old_it)) {
```

Comment:
Don't we need to clear the `set_stale` on failures? Otherwise the flag is get carried over to subsequent commands.

Reply:
I'll check, good catch.

Reply:
I have it resetting to false at the end of do_store_item... that's probably not right since it could error out before calling do_store_item. it's also cleared on conn_new, so that takes care of any case where the connection closes...

Alright, moved it around; forced false in complete_nread_ascii (which called store_etc) and the switch to conn_swallow during conn_nread. Can't switch between bin/ascii without reconnecting.

will keep an eye out for a better way of doing this. Don't want to kill it on every conn_new to avoid the extra memory twiddle for every non-store command :/